src/Asset/PathResolver.php

```diff
@@ -29,11 +41,23 @@ public function resolve(string $path, string $basePath, ThemeInterface $theme):
             $basePathLength = strlen($basePath);
         }
 
-        return sprintf(
-            '%s/_themes/%s/%s',
-            rtrim(substr($path, $basePathPositionAtPath, $basePathLength), '/'),
-            $theme->getName(),
-            ltrim(substr($path, $basePathPositionAtPath + $basePathLength), '/')
-        );
+        $prefixPath = rtrim(substr($path, $basePathPositionAtPath, $basePathLength), '/');
+        $relativePath = trim(substr($path, $basePathPositionAtPath + $basePathLength), '/');
+
+        foreach ($this->assetsProvider->provideDirectoriesForTheme($theme) as $originDir => $targetDir) {
+            $targetDir = trim($targetDir, '/');
+
+            if ($targetDir !== '' && strpos($relativePath, $targetDir) === false) {
+                continue;
+            }
+
+            if (!file_exists($originDir . '/' . str_replace($targetDir, '', $relativePath))) {
```

Comment:
I guess we should create a map of assets during container compilation so that this whole `foreach` could be replaced with a map lookup.